test/e2e/upgrade_test.go

```diff
@@ -0,0 +1,61 @@
+// +build upgrade
+
+package e2e
+
+import (
+	"context"
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"os"
+	"strings"
+	"testing"
+)
+
+const EnvUpdateVersionKey = "UPDATE_TEST_VERSION"
+const upgradeTestTag = "next"
+
+func TestOperatorUpgrade(t *testing.T) {
+
+	upgradeTestVersion := os.Getenv(EnvUpdateVersionKey)
```

Comment:
What happens when the env var is empty?

Reply:
It fails :)

Reply:
When? With which message? Is it easy to understand what's going on?

Reply:
is not, I'll add a validation here to make it more clear.